### ISSUE TITLE
virtualbox-iso: Timeout while waiting for machine to shut down.

### ISSUE THREAD

**Issue by stevenscg** [geerlingguy/packer-centos-7#9]
Hey Jeff,

Did you ever run across an issue like shown below where shutdown fails and the artifact is removed (or never finalized)?

Host: MacOS latest
Guest: CentOS-7 (7.2.1511)

I tried several variations of `shutdown_command`. No change.
I tried a larger `disk_size`. No change.
Increased `shutdown_timeout` to 10m. No change.

What is weird is that I used this same setup and one of your `packer-centos-6` packages just prior to this build attempt. No issues at all.

```
    virtualbox-iso: Installed size: 7.0 M
    virtualbox-iso: Downloading packages:
    virtualbox-iso: Running transaction check
    virtualbox-iso: Running transaction test
    virtualbox-iso: Transaction test succeeded
    virtualbox-iso: Running transaction
    virtualbox-iso: Erasing    : ansible-1.9.4-1.el7.noarch                                   1/1
    virtualbox-iso: Verifying  : ansible-1.9.4-1.el7.noarch                                   1/1
    virtualbox-iso:
    virtualbox-iso: Removed:
    virtualbox-iso: ansible.noarch 0:1.9.4-1.el7
    virtualbox-iso:
    virtualbox-iso: Complete!
    virtualbox-iso: dd: error writing ‘/EMPTY’: No space left on device
    virtualbox-iso: 49367+0 records in
    virtualbox-iso: 49366+0 records out
    virtualbox-iso: 51764068352 bytes (52 GB) copied, 97.2847 s, 532 MB/s
==> virtualbox-iso: Gracefully halting virtual machine...
==> virtualbox-iso: Timeout while waiting for machine to shut down.
==> virtualbox-iso: Unregistering and deleting virtual machine...
==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored: Timeout while waiting for machine to shut down.

==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Timeout while waiting for machine to shut down.

==> Builds finished but no artifacts were created.
```

**Comment by geerlingguy** [comment]
I haven't run into this issue—I just rebuilt all my boxes two days ago, too. Can you reproduce the timeout, or was this just a one-time thing?

**Comment by geerlingguy** [comment]
Also, are you on the latest version of VirtualBox?

**Comment by stevenscg** [comment]
Yup, latest VirtualBox for MacOS: `Version 5.0.10 r104061`

I updated the original description to add one other check I did:
Increased `shutdown_timeout` to 10m. No change.

I can recreate it every time right now with my desired playbook.
I will try to strip the playbook way down and see if I can isolate anything.

**Comment by stevenscg** [comment]
I was able to avoid the issue with a trimmed down playbook!

I'll run a few more build cycles to see if I can isolate anything before closing this.

The playbook that worked is essentially:

```
- hosts: all
  sudo: yes
  gather_facts: yes
  vars:
    # VERY low buffer pool to handle running on small Vagrant-run VMs, etc
    mysql_innodb_buffer_pool_size: 32M

  roles:
    - role: common
      tags: common
    #
    # - role: nfs
    #   tags: build
    #
    # - role: packer-rhel
    #   tags: build
```

Now adding back `nfs` and `packer-rhel`.

**Comment by geerlingguy** [comment]
Sounds good. Something must be limiting the ability of the machine to shut down :(

**Comment by stevenscg** [comment]
FYI, this also worked (using your `nfs` and `packer-rhel` roles)...

```
- hosts: all
  sudo: yes
  gather_facts: yes
  vars:
    # VERY low buffer pool to handle running on small Vagrant-run VMs, etc
    mysql_innodb_buffer_pool_size: 32M

  roles:
    - role: common
      tags: common

    - role: nfs
      tags: build

    - role: packer-rhel
      tags: build
```

**Comment by stevenscg** [comment]
I narrowed it down to the installation of MySQL (client and server). 
Using: Percona Server version: 5.6.27-76.0

Everything else in my standard LEMP playbook worked perfectly.

I took a good build with everything **except** mysql installed, `vagrant up`d to run the mysql roles, logged in, and attempted to halt the VM.

At this point, I was logged out, but the VM didn't shut down.

Trying to `vagrant ssh` in again gives:

```
ssh_exchange_identification: Connection closed by remote host
```

It seems to be hung like this. Still investigating.
